### PR TITLE
Entity and entity models documentations

### PR DESCRIPTION
**Pull request by LambdAurora**
The PR maps some minor stuff, and mainly add new documentations to entity (Axolotl mainly, the idea is to already get some review on that one before doing it on all entities) and entity models (again, not all were done as it's a lot of work and I want feedback before continuing)

The first goal for entity documentation is to explicit the entity relations with the codebase (excluding some specific stuff like axolotl entity calls in the guardian entity class, else it would become too much of a mess)
The rendered result with the axolotl one:
image

I'm still debating if for the tasks I should include all tasks assigned to the axolotl or only entity-specific tasks.

For models it includes a table of the different model parts, using constants when possible:
image

It also adds documentation on some constants, mainly to make javadoc displays their values. Maybe when generating fake sources, if it's a static final field add automatically a `{@value}` if no documentation is present?

### CODE REVIEW

**Comment by Player3324** [comment]
I think it's quite nice for anyone who browses the code/docs or works on related features, this is also a good use for class level Javadoc. It however may not last too long, I expect large refactors towards more data driven or functional approaches, eventually eliminating all(?) relations and model definitions in the above tables. But then said changes have been pending for years, the bulk of 1.17 has been postponed, so maybe it'll stay for quite a while...

**Comment by LambdAurora** [comment]
> I expect large refactors towards more data driven or functional approaches, eventually eliminating all(?) relations and model definitions in the above tables.

For the models I don't expect the tables to change too much? Unless those models classes really do stop existing, the different parts might end up as a set of "absolutely required parts for the model to work", which the tables still can provide in the end.

At least it can be useful for people working with the current code base.

**Comment by Player3324** [comment]
Something derives part states from entity state+time, the moment that bit is data driven the part constraints go away. This is AFAIK what Bedrock does and you mean with model classes, so not unlikely.

I agree that it's useful for the current code and likely years to come, so if you want to invest the time and triage is happy I don't see a reason not to do this.

**Comment by Technici4n** [comment]
Amazing idea! :)

File: mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping [hunk 1]
```diff
@@ -1,4 +1,44 @@
 CLASS net/minecraft/class_5772 net/minecraft/client/render/entity/model/AxolotlEntityModel
+	COMMENT Represents the model of an {@linkplain AxolotlEntity}.
+	COMMENT
+	COMMENT <div class="fabric">
+	COMMENT <table border=1>
+	COMMENT <caption>Model parts of this model</caption>
+	COMMENT <tr>
+	COMMENT   <th>Part Name</th><th>Parent</th><th>Corresponding Field</th>
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping:8]: Hmm, is it possible to squeeze this into only two columns? Like for a top gills, we have it as:
```
<td>{@link #topGills top_gills}</td><td>{@code head}</td>
```
Or use `@code` for parts that aren't in fields. The rendered effect is a code block with a link to that field.

File: mappings/net/minecraft/entity/passive/AxolotlEntity.mapping [hunk 2]
```diff
@@ -1,4 +1,44 @@
 CLASS net/minecraft/class_5762 net/minecraft/entity/passive/AxolotlEntity
+	COMMENT Represents an axolotl, the cutest predator.
```
Comment (liach) [inline mappings/net/minecraft/entity/passive/AxolotlEntity.mapping:2]: :laughing:

File: mappings/net/minecraft/entity/passive/AxolotlEntity.mapping [hunk 3]
```diff
@@ -1,4 +1,44 @@
...
+	COMMENT   <td>Swim Navigation</td><td>{@link AxolotlSwimNavigation}</td>
+	COMMENT </tr>
+	COMMENT <tr>
+	COMMENT   <td>Tasks</td>
+	COMMENT   <td><ul>
+	COMMENT     <li>{@link net.minecraft.entity.ai.brain.task.PlayDeadTask}</li>
+	COMMENT     <li>{@link net.minecraft.entity.ai.brain.task.PlayDeadTimerTask}</li>
+	COMMENT   </ul></td>
+	COMMENT <tr>
+	COMMENT   <td>Variants</td><td>{@link Variant}</td>
```
Comment (liach) [inline mappings/net/minecraft/entity/passive/AxolotlEntity.mapping:32]: I doubt the necessity and usefulness of this particular table; they are quite visible and the tasks are discoverable in the brain class.

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping (hunk 1 above)
Comment (LambdAurora) [reply mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping:8]: My issue with that is it prevents the linking of the key constants.

File: mappings/net/minecraft/entity/passive/AxolotlEntity.mapping (hunk 3 above)
Comment (LambdAurora) [reply mappings/net/minecraft/entity/passive/AxolotlEntity.mapping:32]: They are useful as a "quick navigation" table, maybe also useful for the online javadoc as interactions with the codebase become less obvious.

File: mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/client/render/entity/model/AxolotlEntityModel.mapping:8]: Yeah, only discovered that later.

File: mappings/net/minecraft/entity/passive/AxolotlEntity.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/entity/passive/AxolotlEntity.mapping:32]: Sure. But given the big number of goal tasks, it'd be better if you can move them to the javadoc of the brain class or into a `<details><summary></summary></details>` fold (supported by html 5 which javadoc uses):
```
<details>
<summary>
summary
</summary>

details
</details>
```
effect is like this:
<details>
<summary>
summary
</summary>

details
</details>

File: mappings/net/minecraft/client/model/Model.mapping [hunk 4]
```diff
@@ -1,13 +1,20 @@
 CLASS net/minecraft/class_3879 net/minecraft/client/model/Model
+	COMMENT Represents a dynamic model which has its own render layers and custom rendering.
 	FIELD field_21343 layerFactory Ljava/util/function/Function;
 	METHOD <init> (Ljava/util/function/Function;)V
 		ARG 1 layerFactory
 	METHOD method_23500 getLayer (Lnet/minecraft/class_2960;)Lnet/minecraft/class_1921;
+		COMMENT Returns the render layer for the corresponding texture.
+		COMMENT
+		COMMENT @return the render layer
```
Comment (liach) [inline mappings/net/minecraft/client/model/Model.mapping]:
```suggestion
		COMMENT {@return the render layer for the corresponding texture}
```

File: mappings/net/minecraft/client/render/entity/model/CatEntityModel.mapping [hunk 5]
```diff
@@ -1,5 +1,7 @@
 CLASS net/minecraft/class_3680 net/minecraft/client/render/entity/model/CatEntityModel
 	COMMENT Represents the model of a {@linkplain CatEntity}.
+	COMMENT
+	COMMENT <p>Inherits the model of {@link OcelotEntityModel}.
```
Comment (liach) [inline mappings/net/minecraft/client/render/entity/model/CatEntityModel.mapping]:
```suggestion
	COMMENT <p>Inherits the {@link OcelotEntityModel}.
```

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]
Impressive work!

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**